src/problem.jl

```diff
@@ -25,6 +25,29 @@ function apply(normalization::UniformAdditiveNormalization, x)
     return x .+ normalization.shift
 end
 
+# =====================
+# Control preprocessing
+# =====================
+
+abstract type ControlPreprocessing end
+
+struct NoPreprocessing <: ControlPreprocessing
+end
+
+apply(normalization::NoPreprocessing, x) = x
+
+struct FunctionPreprocessing{F<:Function} <: ControlPreprocessing
+    f::F
+
+    function FunctionPreprocessing(f::F) where {F<:Function}
+        return new{F}(f)
+    end
```

Comment:
This default constructor is always implicitly created and you do not need to add it. But keep it now - we may want to modify it later.